src/record_syscall.cc

```diff
@@ -1715,6 +1716,27 @@ static Switchable prepare_ioctl(RecordTask* t,
       syscall_state.after_syscall_action(record_page_below_stack_ptr);
       return PREVENT_SWITCH;
 
+    // https://github.com/torvalds/linux/blob/254ec036db1123b10e23e1412c191a3cf70dce71/net/bridge/br_ioctl.c#L316-L369
+    case SIOCGIFBR: {
+        auto params = syscall_state.reg_parameter<unsigned long>(3, IN);
+        auto op = t->read_mem(params + 0);
+        switch (op) {
+            // does not mutate memory
+            case BRCTL_GET_VERSION:
+                break;
+            case BRCTL_GET_BRIDGES: {
+                auto len = t->read_mem(params + 2);
+                syscall_state.mem_ptr_parameter(params + 1, len * sizeof(int));
+                break;
+            }
+            case BRCTL_ADD_BRIDGE:
```

Comment:
Like other network devices, these can be tested inside a netns, see e.g. https://github.com/rr-debugger/rr/blob/25212ab087fc22509cb5c951a83966bdc397a45c/src/test/tun.c#L71

Reply:
ooooh, thanks for the pointer! I didn't realize that was available.

Reply:
Done :)